be/src/storage/vectorized/compaction.h

```diff
@@ -44,6 +53,17 @@ class Compaction {
 
     static Status init(int concurreny);
 
+    // choose compaction algorithm according to tablet schema, max columns per group and segment iterator num.
+    // 1. if the number of columns in the schema is less than or equal to max_columns_per_group, use HORIZONTAL.
+    // 2. if source_num is less than or equal to 1, or is more than MAX_SOURCES, use HORIZONTAL.
+    static CompactionAlgorithm choose_compaction_algorithm(size_t num_columns, int64_t max_columns_per_group,
+                                                           size_t source_num);
+
+    static uint32_t get_segment_max_rows(int64_t max_segment_file_size, int64_t input_row_num, int64_t input_size);
+
+    static void get_column_groups(size_t num_columns, size_t num_key_columns, int64_t max_columns_per_group,
```

Comment:
I think you meaning is split column to groups, so `split_column_into_groups` may be a better name.

Reply:
ok